File: batch5pmsushant/myFirebasepractice/app/src/main/java/com/cbitss/myfirebasepractice/RegisterActivity.java
package com.cbitss.myfirebasepractice;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;

import java.util.HashMap;

public class RegisterActivity extends AppCompatActivity {
    EditText numberedit,emailedit,passwordedit,namedit;
    Button signupbtn;
    ProgressBar registerbar;
    FirebaseAuth firebaseAuth;
    FirebaseFirestore db;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);
        connectml();
        firebaseAuth = FirebaseAuth.getInstance();
        db = FirebaseFirestore.getInstance();
        signupbtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String emailtext = emailedit.getText().toString().trim();
                String passorttext = passwordedit.getText().toString().trim();
                String name_text = namedit.getText().toString().trim();
                String numbert_text = numberedit.getText().toString().trim();
                if(TextUtils.isEmpty(emailtext))
                {
                    emailedit.setError("Empty Email");
                }else if(TextUtils.isEmpty(passorttext))
                {
                    passwordedit.setError("Empty password");
                }else if(TextUtils.isEmpty(name_text))
                {
                    namedit.setError("Empty name");
                }else if(TextUtils.isEmpty(numbert_text))
                {
                    numberedit.setError("Empty number");
                }else
                {
                    registerbar.setVisibility(View.VISIBLE);
                    createuser(emailtext,passorttext,numbert_text,name_text);
                }
            }
        });
    }

    private void createuser(String emailtext, String passorttext, String numbert_text, String name_text) {
        firebaseAuth.createUserWithEmailAndPassword(emailtext,passorttext).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
            @Override
            public void onComplete(@NonNull Task<AuthResult> task) {
                if(task.isSuccessful())
                {
                    String uniqueid= firebaseAuth.getCurrentUser().getUid();

                    HashMap<String,String> userdata = new HashMap<>();

                    userdata.put("userEmail",emailtext);
                    userdata.put("userPassword",passorttext);
                    userdata.put("userNumber",numbert_text);
                    userdata.put("userName",name_text);
                    userdata.put("userId",uniqueid);

                    db.collection("AllUser").document(uniqueid).set(userdata).addOnCompleteListener(new OnCompleteListener<Void>() {
                        @Override
                        public void onComplete(@NonNull Task<Void> task) {
                                if(task.isSuccessful()){
                                    registerbar.setVisibility(View.GONE);
                                    Toast.makeText(RegisterActivity.this, "You registered Successfully", Toast.LENGTH_SHORT).show();
                                    startActivity(new Intent(RegisterActivity.this,HomeActivity.class));
                                     finish();

                                }else
                                {
                                    registerbar.setVisibility(View.GONE);
                                    Toast.makeText(RegisterActivity.this, "error "+task.getException(), Toast.LENGTH_SHORT).show();
                                }
                        }
                    });
                }else{
                    registerbar.setVisibility(View.GONE);
                    Toast.makeText(RegisterActivity.this, "error "+task.getException(), Toast.LENGTH_SHORT).show();
                }

            }
        });
    }

    private void connectml() {
        numberedit = findViewById(R.id.numberlayoutregister);
        emailedit = findViewById(R.id.emailayoutregister);
        passwordedit = findViewById(R.id.passwordlayotlayoutregister);
        namedit = findViewById(R.id.namelayoutregister);
        signupbtn = findViewById(R.id.signuptbtn);
        registerbar = findViewById(R.id.registerprogress);
    }
}

File: Rahoul3pm/FirebaseProjects/app/src/main/java/com/batch12pm/firebaseprojects/RegisterActivity.kt
package com.batch12pm.firebaseprojects

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ProgressBar
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore

class RegisterActivity : AppCompatActivity() {
lateinit var numberedit:EditText
lateinit var emailedit:EditText
lateinit var passwordedit:EditText
lateinit var namedit:EditText
lateinit var signupbtn:Button
lateinit var registerbar:ProgressBar
lateinit var firebaseAuth: FirebaseAuth
lateinit var mydb:FirebaseFirestore
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)
        conncetxml()

        firebaseAuth = FirebaseAuth.getInstance()
        mydb = FirebaseFirestore.getInstance()

        signupbtn.setOnClickListener {
            var emailtext = emailedit.text.toString()
            var passwordtext = passwordedit.text.toString()
            var nametext = namedit.text.toString()
            var numbertext = numberedit.text.toString()

            if(TextUtils.isEmpty(emailtext)){
                emailedit.error ="emapty email"
            }else if(TextUtils.isEmpty(passwordtext))
            {
                passwordedit.error = "empty password"
            }else if(TextUtils.isEmpty(nametext))
            {
                namedit.error ="empty name"
            }else if(TextUtils.isEmpty(numbertext))
            {
                numberedit.error="number is empty"
            }else
            {
                registerbar.visibility = View.VISIBLE
                createuser(emailtext,passwordtext,nametext,numbertext)
            }

        }
    }

    private fun createuser(
        emailtext: String,
        passwordtext: String,
        nametext: String,
        numbertext: String
    ) {
                 firebaseAuth.createUserWithEmailAndPassword(emailtext,passwordtext).addOnCompleteListener {task->
                     if(task.isSuccessful)
                     {

                         var userid = firebaseAuth.currentUser!!.uid
                         var usermap:HashMap<String,String> = HashMap()

                         usermap.put("emailText",emailtext)
                         usermap.put("passwordText",passwordtext)
                         usermap.put("nameText",nametext)
                         usermap.put("numberText",numbertext)
                         usermap.put("userid",userid)


                         mydb.collection("FourPmDatabase").document(userid).set(usermap).addOnCompleteListener {task->
                             if(task.isSuccessful){
                                 registerbar.visibility =View.GONE
                                 Toast.makeText(applicationContext,"success create",Toast.LENGTH_SHORT).show()
                                 startActivity(Intent(applicationContext,HomeActivity::class.java))
                                 finish()

                             }else{
                                 registerbar.visibility =View.GONE
                                 Toast.makeText(applicationContext,"error "+task.exception,Toast.LENGTH_SHORT).show()
                             }
                         }

                     }else
                     {
                         registerbar.visibility =View.GONE
                         Toast.makeText(applicationContext,"error "+task.exception,Toast.LENGTH_SHORT).show()
                     }
                 }
    }

    fun conncetxml()
    {
        numberedit = findViewById(R.id.numberlayoutregister);
        emailedit = findViewById(R.id.emailayoutregister);
        passwordedit = findViewById(R.id.passwordlayotlayoutregister);
        namedit = findViewById(R.id.namelayoutregister);
        signupbtn = findViewById(R.id.signuptbtn);
        registerbar = findViewById(R.id.registerprogress);
    }
}